### PR TITLE
Show panel back to the last position before hiding it

### PR DESCRIPTION
**Pull request by paulnicolet**
Hi 👋,
I met the use case for a project:
The idea is to add a new behaviour allowing to **show the panel back to the last position it was right before hiding it**, instead of always use the layout initial position.

Let me know if some improvements are needed, or if this is not a welcome behaviour, which is fine 😉

Paul

### CODE REVIEW

**Comment by hmoreau94** [comment]
Please accept the pull request :)

**Comment by SCENEE** [comment]
Thank you for your great PR! I agree to support the use case. Please check my comments as below?

**Comment by paulnicolet** [comment]
Hi ! Nice 😄

> Thank you for your great PR! I agree to support the use case. Please check my comments as below?

Sorry but which comments ?

**Comment by SCENEE** [comment]
Sorry, @paulnicolet. I hadn't submitted my comment completely. Now you can see my comment as  above.

File: Framework/Sources/FloatingPanelBehavior.swift [hunk 1]
```diff
@@ -68,6 +68,12 @@ public protocol FloatingPanelBehavior {
     ///
     /// This method allows the behavior to activate the rubber band effect to a given edge of the surface view. By default, the effect is disabled.
     func allowsRubberBanding(for edge: UIRectEdge) -> Bool
+    
+    /// Decide if showing the panel should use the last position before hiding if possible.
+    /// If false, it will always use the layout initial position.
+    ///
+    /// Default is false, which will always use the initial position.
+    var showToLastPositionIfPossible: Bool { get }
```
[the remark this answers is not in the document]
Comment (headstream12) [reply Framework/Sources/FloatingPanelBehavior.swift:76]: Hello. How I can set this property to true?

[the remark this answers is not in the document]
Comment (SCENEE) [reply Framework/Sources/FloatingPanelBehavior.swift:76]: Hello. You can define a behavior object adopting `FloatingPanelBehavior` which has the readonly property or readwrite property of `shouldRestorePositionBeforeHiding` as below  😌 
```swift
class FloatingPanelStocksBehavior: FloatingPanelBehavior {
    var shouldRestorePositionBeforeHiding: Bool = true
    // Or
    // var shouldRestorePositionBeforeHiding: Bool { true }
}
```

[the remark this answers is not in the document]
Comment (SCENEE) [reply Framework/Sources/FloatingPanelBehavior.swift:76]: By the way I would like to merge this PR after renaming this property name, but do I need to do anything?

[the remark this answers is not in the document]
Comment (Jamonek) [reply Framework/Sources/FloatingPanelBehavior.swift:76]: I'd like to update the property name. Hmm

**Comment by scenee** [comment]
Sorry for the delay in my response. This PR was created for v1, and since I will not release a new version with v1, I have closed this PR. As an alternative, you can always keep the FloatingPanelLayout after hiding it and update its `initialState` on `floatingPanelWillRemove` to get the expected behavior 😉

**Pull request closed without merging**